lib/nx_zlib.h

```diff
@@ -181,15 +181,12 @@ struct nx_dev_t {
 	int use_cnt;       /* total number of users */
 	pid_t creator_pid; /* PID of the process that allocated this handle */
 
-	/* https://github.com/sukadev/linux/blob/vas-kern-v8.1/tools/testing/selftests/powerpc/user-nx842/compress.c#L514 */
-	struct {
-		int16_t version;
-		int16_t id;
-		int64_t flags;
-		void *paste_addr;
-		int fd;
-		void *vas_handle;
-	}; /* vas */
+	int16_t version;
+	int16_t id;
+	int64_t flags;
+	void *paste_addr;
+	int fd;
+	int function;
```

Comment:
Would be good to have a comment in this block indicating that this is the vas handle.

Reply:
But the "vas handle" does not exist anymore. This struct is our handle to communicate with NX, and past_addr, fd and function are just parts of that NX handle. I don't see how keeping the distinction is useful.

Reply:
I think it would be useful for new developers. The `version`, `id`, and `flags` fields are vas specific attributes to be used on ioctl, and `past_addr` is the vas memory address.
`fd` is actually the device file descriptor so that and `function` are not directly part of vas.
Maybe not keeping a distinction between what is vas related and the rest but at least adding comments describing the fields should help.

Reply:
@rzinsly Done. I added comments for each field and ended up removing `version`, `id` and `flags`. `open_device_nodes` just uses hardcoded values for `version` and `flags` when doing the ioctl, and `id` is given as parameter to that function, so the fields in this struct are not being used.

Reply:
@mscastanho Perfect!
LGTM, thanks!